src/uct/ib/dc/dc_mlx5.c

```diff
@@ -776,9 +776,20 @@ ucs_status_t uct_dc_device_query_tl_resources(uct_ib_device_t *dev,
 
 static inline ucs_status_t uct_dc_mlx5_iface_flush_dcis(uct_dc_mlx5_iface_t *iface)
 {
+    uct_dc_mlx5_ep_t *ep;
     int i;
 
     for (i = 0; i < iface->tx.ndci; i++) {
+        if (!uct_dc_mlx5_iface_is_dci_rand(iface)) {
```

Comment:
can we have a global counter (per iface) of how many are waiting for grant?
also, why checking uct_dc_mlx5_ep_fc_wait_for_grant() and not FC window? maybe add assertion that if FC window is empty, we are waiting for grant?

Reply:
I think global counter per iface is a good idea, because it will solve the problem for all TX policy types